libevmasm/Assembly.cpp

```diff
@@ -674,15 +674,18 @@ LinkerObject const& Assembly::assemble() const
 		case AssignImmutable:
 			for (auto const& offset: immutableReferencesBySub[i.data()].second)
 			{
-				ret.bytecode.push_back(uint8_t(Instruction::DUP1));
+				ret.bytecode.push_back(uint8_t(Instruction::DUP2));
+				ret.bytecode.push_back(uint8_t(Instruction::DUP2));
 				// TODO: should we make use of the constant optimizer methods for pushing the offsets?
 				bytes offsetBytes = toCompactBigEndian(u256(offset));
 				ret.bytecode.push_back(uint8_t(Instruction::PUSH1) - 1 + offsetBytes.size());
 				ret.bytecode += offsetBytes;
+				ret.bytecode.push_back(uint8_t(Instruction::ADD));
```

Comment:
Is this optimized away?

Reply:
No, I don't think so. That's why we didn't do this when we first introduced all of this...

Reply:
And why I'm still wondering, whether this is a good idea or if we could do better :-).